packages/application-shell/src/components/redirector/redirector.tsx

```diff
@@ -9,29 +9,41 @@ import { location } from '../../utils/location';
 type QueryParams = {
   reason?: typeof LOGOUT_REASONS[keyof typeof LOGOUT_REASONS];
   redirectTo?: string;
+  client_id?: string;
+  response_type?: string;
+  scope?: string;
+  state?: string;
+  nonce?: string;
 };
 type Props = {
   to: string;
+  origin?: string;
   location?: RouteComponentProps['location'];
-  queryParams: QueryParams;
+  queryParams?: QueryParams;
 };
 
-const redirectTo = (targetUrl: string) => location.replace(targetUrl);
+const redirectTo = (targetUrl: string) => {
+  location.replace(targetUrl);
+};
 
 const Redirector = (props: Props) => {
-  React.useEffect(() => {
```

Comment:
We don't need all those variables to be defined inside the `useEffect`, hence this change.